spec/controllers/queue_items_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'spec_helper'
+describe QueueItemsController do
+  let(:user1) {Fabricate(:user)}
+  let(:queue_item1) {Fabricate(:queue_item, user: user1)}
+  let(:queue_item2) {Fabricate(:queue_item, user: user1)}
+  let(:video1) {Fabricate(:video)}
+  let(:video2) {Fabricate(:video)}
+  let(:video3) {Fabricate(:video)}
+
+  context "logged in user" do 
+    before do
+      cookies.signed[:auth_token] = user1.auth_token
+    end
+
+    describe "GET index" do
+      before do
+        get :index, user_id: user1.id
+      end
+
+      it "should assign all of current users's queue items to queue" do
+        expect(assigns(:queue_items)).to match_array([queue_item1,queue_item2])
+      end
+      
+      it "should render queue index page" do
+        expect(response).to render_template :index
+      end
+    end
+
+    describe "POST create" do
+      context "empty queue" do
+        before do
+          post :create, user_id: user1.id, video_id: video1.id 
+        end
+
+        it "should create and save queue_item" do
+          expect(user1.queue_items.count).to eq(1)
+        end
+
+        it "should create queue item with position 1 if first queue item" do
+          expect(user1.queue_items.first.position).to eq(1)
+        end
+      end
+
+      context "non empty queue" do
+        before do 
+          queue_item4 = Fabricate(:queue_item, user: user1, position: user1.queue_items.size + 1, video: video1)
```

Comment:
Local variables in `before` blocks are only available within that block. You don't need to assign `queue_item4` and `queue_item5`.